pyro/infer/autoguide/guides.py

```diff
@@ -182,12 +178,17 @@ def forward(self, *args, **kwargs):
             self._setup_prototype(*args, **kwargs)
 
         # create all plates
-        self.plates = {frame.name: pyro.plate(frame.name, frame.size, dim=frame.dim)
-                       for frame in sorted(self._plates.values())}
+        with poutine.block():
```

Comment:
Can you explain the need for `poutine.block` here? As I understand, autoguides do not currently support subsampling, and hence the `_Subsample` sites can be ignored; however if we ever were to support subsampling I believe we would need to not block here.

Reply:
I noticed this when working with AutoGuideList where the model has a `plate` statement. In that case, when we generate the guide trace there are two sample calls for the `plate` site - one from `create_plates` and the other in the `__call__`. I think the plate context should be correctly captured in `result.updatepart(*args, **kwargs)`.

Reply:
I think this should actually be
```py
# create all plates
self.plates = self._create_plates()
```

Reply:
Or maybe we should update `AutoGuide._create_plates()` as follows:
```py
class AutoGuide(PyroModule):
    def _update_plates(self):
        if self.master is None:
            self.plates = {frame.name: ...}
        else:
            self.plates = self.master().plates
        return self.plates
````